portfolio/src/main/webapp/markermap.js

```diff
@@ -124,23 +125,50 @@ class MarkerMap {
     fetch('/markers')
       .then(response => response.json())
       .then((markers) => {
+        for(let marker of this.markers) {
+          marker.setMap(null);
+        }
```

Comment:
You are clearing the individual markers from the map, but not removing them from `this.markers` which means the array will just grow very big with useless null markers.